tldw_Server_API/app/core/StudySuggestions/topic_pipeline.py

```
@@ -1,65 +1,173 @@
+        normalized_group_rows.append((namespace, semantic_canonical, semantic_canonical, raw_labels))
 
-    normalized: list[NormalizedTopicLabel] = []
-    for entries in grouped.values():
-        canonical = min((cleaned for cleaned, _ in entries), key=_canonical_preference)
-        normalized.append(
+    semantic_label_counts: dict[str, int] = {}
+    for _, semantic_canonical, _, _ in normalized_group_rows:
+        semantic_label_counts[semantic_canonical] = semantic_label_counts.get(semantic_canonical, 0) + 1
+
+    normalized_topics: list[NormalizedTopicLabel] = []
+    for namespace, semantic_canonical, representative_canonical, raw_labels in normalized_group_rows:
```

Ruff (0.15.9)

[warning] Loop control variable `representative_canonical` not used within loop body

Rename unused `representative_canonical` to `_representative_canonical`

(B007)